### DISCUSSION TITLE
Remove-GitHubRepository -Uri value

### DISCUSSION THREAD

**Discussion by Liturgist** [microsoft/PowerShellForGitHub#393] [Q&A]
Remove-GitHubRepository has a `-Uri` parameter.

Get-GitHubRepository returns an object with no "uri" member names; they are all "url". I used the `RepositoryUrl` member successfully. Is there another "url" member that would be better to use with `Remove-GitHubRepository`?

Is it possible that `-Uri` should be an alias for `-RepositoryUrl`?

Is there a technical reason for calling it a `uri` rather than a `url`?

**Comment by HowardWolosky** [comment]
A URL is a specific type of a URI.  There are many discussions on the differences and similarities between them.  [Here's one such resource](https://danielmiessler.com/study/difference-between-uri-url/). 

`RepositoryUrl` is the correct property on any of the module's GitHub objects to use for that parameter.  But generally speaking, you shouldn't even need to think about it/access it directly, as you can just pipe the relevant object into the desired function and it'll grab it automatically.  `RepositoryUrl` was introduced throughout the entire module as part of #242 when I added Pipeline support.  I needed a consistent name for that parameter that I could add to every object so that all functions could correctly parse a GitHub object, given that there was a chance that `Uri` might end up existing in some objects coming from GitHub with a different meaning.  At the time, it felt less of a "breaking change" to keep the existing name of the parameter and add an alias that satisfied the needs of the new objects.  In retrospect, it would have been fine either way.

At this point I could likely be convinced that the actual name of the parameter should be `RepositoryUrl` with an alias of `Uri` to keep it from breaking anything -- it just feels like a lot of work updating all the code and documentation for what would eventually result in absolutely no end-user benefit/change.  I'm not that incentivized to do it, but I'd be open to someone else doing it if they were exhaustive in their update.
